openedx/core/djangoapps/profile_images/views.py

```diff
@@ -105,9 +108,24 @@ class ProfileImageView(APIView):
           the user exists or not.
     """
 
-    parser_classes = (MultiPartParser, FormParser)
+    parser_classes = (MultiPartParser, FormParser, TypedFileUploadParser)
     authentication_classes = (OAuth2AuthenticationAllowInactiveUser, SessionAuthenticationAllowInactiveUser)
     permission_classes = (permissions.IsAuthenticated, IsUserInUrl)
+    _supported_media_types = None
+
+    @property
+    def supported_media_types(self):
```

Comment:
I put in in the view for a few reasons: 

1. The acceptable file types are defined in profile_images.images.IMAGE_TYPES.  I don't want the parser to depend on a global variable in an unrelated subpackage.  
2. The set of acceptable image types is actually defined by what the view and its related helper functions are willing to process.  If we add SVG support at some point in the future, we should only have to add that at the image processing layer.  We shouldn't have to modify the parser as well (Don't Repeat Yourself).  
3. The parser is Typed, in that it only allows the types that the view wants to accept.  This could be different for different views depending on how they are using it. Therefore, I want the set of accepted types to be provided by the view.  DRF does not give us the capability to instantiate the parser with different values for different uses.  It does, however, pass the view to the parse method, so this is a clean way to pass in that information that doesn't require fiddling with globals or subclassing the parser for every combination of types you want to support.  The *view* supports jpeg, png, and gif, so the parser allows them.

The piece that feels ugly to me is that while the view *should* specify the types that it accepts, it should not have to know how to map those types to file extensions.  I included that because it was easy, and already available on the `IMAGE_TYPES` global.  I would prefer to have the parser know how to map any media type to its acceptable file extensions, and the view only pass in the set of media types it wants to support.  Do you know if such a mapping is readily available somewhere?  From a library, or a csv file somewhere that we could use?

I suppose if we don't have that information, I could create a partial mapping on the parser, and if it receives a mime-type that isn't included, it would skip the extension check.  That seems like reasonable fall-back behavior anyway.  Then you could also use the parser without specifying supported_media_types, in which case it would allow any media types, but could still validate the extensions for the ones it knows about.